tests/unit/s2n_x509_validator_test.c

```diff
@@ -1365,6 +1365,76 @@ int main(int argc, char **argv)
         s2n_x509_trust_store_wipe(&trust_store);
     };
 
+    /* test validator in safe mode, with default host name validator. Connection server matches the IPv6 address on the certificate. */
+    {
+        struct s2n_x509_trust_store trust_store;
+        s2n_x509_trust_store_init_empty(&trust_store);
+        EXPECT_EQUAL(0, s2n_x509_trust_store_from_ca_file(&trust_store, S2N_IP_V6_LO_RSA_CERT, NULL));
+
+        struct s2n_x509_validator validator;
+        s2n_x509_validator_init(&validator, &trust_store, 1);
+
+        struct s2n_connection *connection = s2n_connection_new(S2N_CLIENT);
+        EXPECT_NOT_NULL(connection);
+
+        /* the provided hostname should be an empty string */
+        struct host_verify_data verify_data = { .callback_invoked = 0, .found_name = 0, .name = "::1\0" };
+        EXPECT_SUCCESS(s2n_connection_set_verify_host_callback(connection, verify_host_verify_alt, &verify_data));
+
+        DEFER_CLEANUP(struct s2n_stuffer cert_chain_stuffer = { 0 }, s2n_stuffer_free);
+        EXPECT_OK(
+                s2n_test_cert_chain_data_from_pem(connection, S2N_IP_V6_LO_RSA_CERT, &cert_chain_stuffer));
```

Comment:
Nit: I hate this formatting D: The newline only makes it two characters shorter!

Reply:
You think I like it? That's what `clang-format` did to it.

Reply:
I think Harrison found that he could manually change line breaks like that. Clang-format makes terrible choices when wrapping, but will accept less terrible choices.